sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/resource/LambdaResource.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.aws.resource;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.attributes.SemanticAttributes;
+import io.opentelemetry.sdk.resources.ResourceProvider;
+import java.util.Map;
+
+/** A {@link ResourceProvider} which provides information about the AWS Lambda function. */
+public final class LambdaResource extends ResourceProvider {
+  private final Map<String, String> environmentVariables;
+
+  public LambdaResource() {
+    this(System.getenv());
+  }
+
+  // Visible for testing
+  LambdaResource(Map<String, String> environmentVariables) {
+    this.environmentVariables = environmentVariables;
+  }
+
+  @Override
+  protected Attributes getAttributes() {
+    if (!isLambda()) {
+      return Attributes.empty();
+    }
+    String region = environmentVariables.getOrDefault("AWS_REGION", "");
+    String functionName = environmentVariables.get("AWS_LAMBDA_FUNCTION_NAME");
+    String functionVersion = environmentVariables.getOrDefault("AWS_LAMBDA_FUNCTION_VERSION", "");
+
+    AttributesBuilder builder =
+        Attributes.builder()
+            .put(SemanticAttributes.CLOUD_PROVIDER, SemanticAttributes.CloudProviderValues.AWS)
+            .put(SemanticAttributes.FAAS_NAME, functionName);
```

Comment:
Shouldn't we check empty for this too? It's not clear why we check for the others only but not this one.

Reply:
This one cannot be empty, it's being checked in the `isLambda()` function.

Reply:
I refactored it a bit so that all env vars are retrieved in exactly the same way.